plugins/AtlasExplorer/io/nrrdloader/data/decoders/AsciiDecoder.cpp

```diff
@@ -34,13 +33,16 @@ class AsciiParser
 {
 public:
     template<typename T>
-    static std::vector<T> parse(const std::vector<std::string_view> &tokens)
+    static std::vector<T> parse(std::string_view input, size_t tokenCount)
     {
-        std::vector<T> result(tokens.size());
-
-        for (size_t i = 0; i < tokens.size(); ++i)
+        std::vector<T> result;
+        result.reserve(tokenCount);
+        while (!input.empty())
         {
-            brayns::StringParser<T>::parse(tokens[i], result[i]);
+            auto token = brayns::StringExtractor::extractUntilOneOf(input, brayns::StringConstants::tokenDelimiters);
```

Comment:
extractToken ?

Reply:
extractToken extracts only space-separated tokens, whereas the nrrd files have more delimiters. And I dont think its correct to use the list of delimiters from a specific file format into a generic function that is opaque to the user, even more when there is API like extractUntilOneOf which does what its needed.